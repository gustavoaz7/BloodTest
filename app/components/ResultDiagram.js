import React from 'react';
import { View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import styled from 'styled-components/native';
import DiagramLabel from './DiagramLabel';
import { testResultType } from '../types';

export default function ResultDiagram({ result }) {
  return (
    <>
      <BarContainer>
        {result.map(({ color, percentage }, index) => (
          <Bar key={`${color}-${percentage}`}>
            {index !== 0 && <Separator />}
            <BarSection color={color} percentage={percentage} />
          </Bar>
        ))}
      </BarContainer>
      <LabelBox>
        {result.map(resultItem => (
          <DiagramLabel
            key={`${resultItem.biomarker}-${resultItem.color}`}
            style={styles.diagramLabel}
            result={resultItem}
          />
        ))}
      </LabelBox>
    </>
  );
}

const styles = StyleSheet.create({
  diagramLabel: {
    width: '50%',
  },
});

const Row = styled(View)`
  flex-direction: row;
`;

const Bar = styled(Row)`
  flex: 1;
`;

const BarContainer = styled(View)`
  flex-direction: row;
  height: 16px;
  border-radius: 8px;
  overflow: hidden;
  margin-bottom: 8px;
`;

const Separator = styled(View)`
  width: 1px;
`;

const BarSection = styled(View)`
  ${({ color }) => `background-color: ${color}`};
  ${({ percentage }) => `flex: ${percentage}`};
`;

const LabelBox = styled(Row)`
  flex: 1;
  flex-wrap: wrap;
`;

ResultDiagram.propTypes = {
  result: PropTypes.arrayOf(testResultType).isRequired,
};
