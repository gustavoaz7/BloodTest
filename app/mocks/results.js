export const resultsMock = [
  {
    id: '0b82fed2-a091-4318-ba02-4d5e6192d6f7',
    name: 'Citlalli Kuhic',
    email: 'Berniece.Haley42@hotmail.com',
    samples: [
      {
        time: 'morning',
        value: 0.48,
      },
      {
        time: 'afternoon',
        value: 0.39,
      },
      {
        time: 'evening',
        value: 0.19,
      },
    ],
  },
  {
    id: 'acf65e6a-1878-4cd9-bf6f-440dafd4fa18',
    name: 'Rachael Lehner',
    email: 'Teagan_Ratke83@gmail.com',
    samples: [
      {
        time: 'morning',
        value: 0.67,
      },
      {
        time: 'afternoon',
        value: 0.31,
      },
      {
        time: 'evening',
        value: 0.24,
      },
    ],
  },
  {
    id: 'ca285be1-0d6b-42f6-bfec-b3aeaa2aaf8f',
    name: 'Willa Quitzon',
    email: 'Palma17@yahoo.com',
    samples: [
      {
        time: 'morning',
        value: 0.73,
      },
      {
        time: 'afternoon',
        value: 0.19,
      },
      {
        time: 'evening',
        value: 0.14,
      },
    ],
  },
  {
    id: '95eba822-a7b3-4938-8c94-005733701c1f',
    name: 'Miss Audie Moore',
    email: 'Giuseppe19@hotmail.com',
    samples: [
      {
        time: 'morning',
        value: 0.26,
      },
      {
        time: 'afternoon',
        value: 0.09,
      },
      {
        time: 'evening',
        value: 0.29,
      },
    ],
  },
  {
    id: '4fea89aa-bcf7-42b0-a5e8-202fc03a02fe',
    name: 'Yesenia Nicolas',
    email: 'Maeve_Jacobs14@yahoo.com',
    samples: [
      {
        time: 'morning',
        value: 0.78,
      },
      {
        time: 'afternoon',
        value: 0.26,
      },
      {
        time: 'evening',
        value: 0.24,
      },
    ],
  },
  {
    id: '17d11f05-2580-49b5-bb17-b786209c4792',
    name: 'Ms. Harmon Kunze',
    email: 'Johnson90@yahoo.com',
    samples: [
      {
        time: 'morning',
        value: 0.47,
      },
      {
        time: 'afternoon',
        value: 0.09,
      },
      {
        time: 'evening',
        value: 0.09,
      },
    ],
  },
];
