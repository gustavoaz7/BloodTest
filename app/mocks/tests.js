export const testsMock = [
  {
    id: '5d41b693ed44f7ee9f9436d0',
    title: 'Healthcare',
    date: '17.07.2019',
    result: [
      {
        biomarker: 'fats',
        color: '#e30613',
        percentage: 25,
      },
      {
        biomarker: 'carbohydrates',
        color: '#D3235D',
        percentage: 25,
      },
      {
        biomarker: 'vitamins',
        color: '#547F81',
        percentage: 25,
      },
      {
        biomarker: 'magnesium',
        color: '#0085FF',
        percentage: 25,
      },
    ],
  },
  {
    id: '5d41b6938f1360cdde5ecc89',
    title: 'Advanced',
    date: '08.07.2019',
    result: [
      {
        biomarker: 'protein',
        color: '#E2D669',
        percentage: 25,
      },
      {
        biomarker: 'vitamins',
        color: '#ffcc00',
        percentage: 25,
      },
      {
        biomarker: 'magnesium',
        color: '#e30613',
        percentage: 25,
      },
      {
        biomarker: 'carbohydrates',
        color: '#0085FF',
        percentage: 25,
      },
    ],
  },
  {
    id: '5d41b693084f4f55217fd864',
    title: 'Sport',
    date: '30.07.2019',
    result: [
      {
        biomarker: 'fiber',
        color: '#E2D669',
        percentage: 25,
      },
      {
        biomarker: 'fats',
        color: '#D3235D',
        percentage: 25,
      },
      {
        biomarker: 'magnesium',
        color: '#547F81',
        percentage: 25,
      },
      {
        biomarker: 'vitamins',
        color: '#0085FF',
        percentage: 25,
      },
    ],
  },
];
