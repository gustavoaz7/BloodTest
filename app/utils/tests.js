export const flushPromises = () => new Promise(setImmediate);

export const stackNavigationProps = {
  state: {},
  dispatch() {},
  goBack() {},
  dismiss() {},
  navigate() {},
  getParam() {},
  setParams() {},
  emit() {},
  addListener() {},
  isFocused() {},
  isFirstRouteInParent() {},
  dangerouslyGetParent() {},
  router() {},
};
