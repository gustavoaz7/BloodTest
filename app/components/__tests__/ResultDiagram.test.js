import React from 'react';
import { shallow } from 'enzyme';
import ResultDiagram from '../ResultDiagram';
import { testsMock } from '../../mocks/tests';

describe('ListItem', () => {
  const { result } = testsMock[0];

  it('renders without crashing', () => {
    shallow(<ResultDiagram result={result} />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<ResultDiagram result={result} />);
    expect(wrapper).toMatchSnapshot();
  });
});
