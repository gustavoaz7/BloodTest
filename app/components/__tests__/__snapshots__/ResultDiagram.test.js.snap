// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem matches snapshot 1`] = `
Array [
  <Styled(View)
    key="fats-#e30613"
  >
    <Styled(View)>
      <Text>
        fats
      </Text>
    </Styled(View)>
    <Styled(View)>
      <Text>
        25%
      </Text>
      <Styled(View)>
        <Styled(View)
          color="#e30613"
          percentage={35}
        />
      </Styled(View)>
    </Styled(View)>
  </Styled(View)>,
  <Styled(View)
    key="carbohydrates-#D3235D"
  >
    <Styled(View)>
      <Text>
        carbohydrates
      </Text>
    </Styled(View)>
    <Styled(View)>
      <Text>
        25%
      </Text>
      <Styled(View)>
        <Styled(View)
          color="#D3235D"
          percentage={35}
        />
      </Styled(View)>
    </Styled(View)>
  </Styled(View)>,
  <Styled(View)
    key="vitamins-#547F81"
  >
    <Styled(View)>
      <Text>
        vitamins
      </Text>
    </Styled(View)>
    <Styled(View)>
      <Text>
        25%
      </Text>
      <Styled(View)>
        <Styled(View)
          color="#547F81"
          percentage={35}
        />
      </Styled(View)>
    </Styled(View)>
  </Styled(View)>,
  <Styled(View)
    key="magnesium-#0085FF"
  >
    <Styled(View)>
      <Text>
        magnesium
      </Text>
    </Styled(View)>
    <Styled(View)>
      <Text>
        25%
      </Text>
      <Styled(View)>
        <Styled(View)
          color="#0085FF"
          percentage={35}
        />
      </Styled(View)>
    </Styled(View)>
  </Styled(View)>,
]
`;
